optimum/intel/ipex/inference.py

```diff
@@ -20,6 +22,51 @@
     import intel_extension_for_pytorch as ipex
 
 
+def prepare_jit_inputs(inputs, model_pipeline, dtype):
+    input_length = 1 if isinstance(inputs, str) else len(inputs)
+    if model_pipeline.tokenizer.pad_token is None:
+        model_pipeline.tokenizer.pad_token = model_pipeline.tokenizer.eos_token
+    dummy_input = model_pipeline.tokenizer.batch_encode_plus(inputs, return_tensors="pt", padding=True)
+    normalized_config = NormalizedConfigManager.get_normalized_config_class(model_pipeline.model.config.model_type)(
+        model_pipeline.model.config
+    )
+    num_block_layers = normalized_config.num_layers
+    num_attention_heads = normalized_config.num_attention_heads
+    num_embedding_size_per_head = normalized_config.hidden_size // normalized_config.num_attention_heads
+    if model_pipeline.model.config == "bloom":
+        past_key_values = tuple(
+            (
+                torch.zeros(int(num_attention_heads * input_length), num_embedding_size_per_head, 1).to(dtype),
+                torch.zeros(int(num_attention_heads * input_length), 1, num_embedding_size_per_head).to(dtype),
+            )
+            for _ in range(num_block_layers)
+        )
+    else:
+        past_key_values = tuple(
+            (
+                torch.zeros(input_length, num_attention_heads, 1, num_embedding_size_per_head).to(dtype),
+                torch.zeros(input_length, num_attention_heads, 1, num_embedding_size_per_head).to(dtype),
+            )
+            for _ in range(num_block_layers)
+        )
+    dummy_input["attention_mask"] = torch.cat(
+        [
+            torch.zeros(dummy_input["attention_mask"].shape[0], 1).to(dummy_input["attention_mask"].dtype),
+            dummy_input["attention_mask"],
+        ],
+        -1,
+    )
+    if model_pipeline.task == "text-generation" and model_pipeline.model.config.use_cache:
+        jit_inputs = (dummy_input["input_ids"], past_key_values, dummy_input["attention_mask"])

```

Comment:
The parameter order can very depending on the modeling definition, a more robust way to create the dummy inputs would be to generate them using a configuration such as done for the optimum ONNX [export](https://github.com/huggingface/optimum/blob/v1.7.1/optimum/exporters/onnx/convert.py#L208).

Reply:
It seems great, I can have a try. BTW, do you have any examples of how to use it? Thanks!